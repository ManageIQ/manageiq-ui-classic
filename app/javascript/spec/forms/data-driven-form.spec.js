import React from 'react';
import { mount } from 'enzyme';
import { componentTypes } from '@@ddf';
import { act } from 'react-dom/test-utils';

import MiqFormRenderer from '../../forms/data-driven-form';

describe('DataDrivenForm', () => {
  let initialProps;

  beforeEach(() => {
    const schema = {
      fields: [{
        component: componentTypes.TEXT_FIELD,
        name: 'name',
        labelText: 'label',
      }],
    };
    initialProps = {
      schema,
      store: ManageIQ.redux.store,
    };
  });

  it('should render correctly', () => {
    const wrapper = mount(<MiqFormRenderer {...initialProps} />);
    expect(wrapper.find('form')).toHaveLength(1);
    expect(wrapper.find('input')).toHaveLength(1);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should set pristine in reducer when changing state', () => {
    const wrapper = mount(<MiqFormRenderer {...initialProps} />);
    const changeInput = (value) => act(() => {
      wrapper.find('input').first().simulate('change', { target: { value } });
    });

    changeInput('changed-value');

    expect(ManageIQ.redux.store.getState().FormButtons.pristine).toEqual(false);

    changeInput('');

    expect(ManageIQ.redux.store.getState().FormButtons.pristine).toEqual(true);
  });
});
