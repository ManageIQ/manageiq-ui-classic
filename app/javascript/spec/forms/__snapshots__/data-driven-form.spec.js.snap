// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataDrivenForm should render correctly 1`] = `"<form class=\\"bx--form form-0-2-3 form-react\\" novalidate=\\"\\"> <div class=\\"bx--form-item bx--text-input-wrapper\\"><label for=\\"name\\" class=\\"bx--label\\">label</label><div class=\\"bx--text-input__field-outer-wrapper\\"><div class=\\"bx--text-input__field-wrapper\\"><input id=\\"name\\" type=\\"text\\" class=\\"bx--text-input bx--text-input--md\\" name=\\"name\\" value=\\"\\"></div></div></div><div class=\\"bx--btn-set\\"><button variant=\\"primary\\" tabindex=\\"0\\" class=\\"bx--btn bx--btn--primary bx--btn--disabled\\" disabled=\\"\\" type=\\"submit\\">Save</button></div> </form>"`;
